src/main/java/gregicadditions/machines/MetaTileEntityMachineHolder.java

```diff
@@ -0,0 +1,95 @@
+import gregtech.api.metatileentity.ITieredMetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.api.metatileentity.multiblock.IMultiblockAbilityPart;
+import gregtech.api.metatileentity.multiblock.MultiblockAbility;
+import gregtech.common.metatileentities.electric.multiblockpart.MetaTileEntityItemBus;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemStackHandler;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class MetaTileEntityMachineHolder extends MetaTileEntityItemBus implements IMultiblockAbilityPart<IItemHandlerModifiable> {
+
+    protected IItemHandlerModifiable machineItemHandler;
+
+    public MetaTileEntityMachineHolder(ResourceLocation metaTileEntityId) {
+        super(metaTileEntityId, 0, false);
+        machineItemHandler = new MachineImportItemHandler();
+        initializeInventory();
+    }
+
+    @Override
+    public MetaTileEntity createMetaTileEntity(MetaTileEntityHolder holder) {
+        return new MetaTileEntityMachineHolder(metaTileEntityId);
+    }
+
+    @Override
+    public MultiblockAbility<IItemHandlerModifiable> getAbility() {
+        return GACapabilities.PA_MACHINE_CONTAINER;
+    }
+
+    @Override
+    public void registerAbilities(List<IItemHandlerModifiable> abilityList) {
+        abilityList.add(machineItemHandler);
+    }
+
+    @Override
+    public IItemHandlerModifiable getImportItems() {
+        return machineItemHandler;
+    }
+
+    @Override
+    public IItemHandler getItemInventory() {
+        return machineItemHandler;
+    }
+
+    @Override
+    protected IItemHandlerModifiable createImportItemHandler() {
+        return machineItemHandler;
+    }
+
+    @Override
+    protected boolean openGUIOnRightClick() {
+        TileEntityProcessingArray controller = (TileEntityProcessingArray) getController();
+        if(controller != null && controller.getWorkable().isActive()) {
+            return false;
+        }
+        return true;
+    }
+
+    private static class MachineImportItemHandler extends ItemStackHandler {
+
+        @Nonnull
+        @Override
+        public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate) {
+
+            if(!isItemValid(slot, stack)) {
+                return stack;
+            }
+
+            return super.insertItem(slot, stack, simulate);
+        }
+
+        @Override
+        public boolean isItemValid(int slot, @Nonnull ItemStack stack) {
+
+            if(stack.getItem() instanceof MachineItemBlock) {
+                MetaTileEntity mte = MachineItemBlock.getMetaTileEntity(stack);
+
+                return mte instanceof ITieredMetaTileEntity;
+            }
+
+            return false;
+        }
+
+    }
```

Comment:
might want to override `extractItem` here to prevent removal of machines while the PA is active (re: the comment about "locking" from the previous review)

Reply:
This is a good idea, which I think I will do. I just have to figure out how to access the PA's state from within the `MachineImportHandler` class

Reply:
I am now remembering why I did not override `extractItem`. I cannot figure out a way to access the state of the Processing Array in this method

Reply:
The one issue that I am having with implementing this, is that the multiblock's active state is not static, therefore it is not possible to reference the active state inside an override to `extractItem`. 

I only want to prevent item extraction when the PA is active, and not in any other case, however I cannot figure out a way to get this active state to check in the `extractItem` method.

Reply:
you just need the `MachineImportItemHandler` to be able to get ahold of the PA controller instance? then all you need to do is give it a reference to its owning `MetaTileEntityMachineHolder` (e.g. by passing it in the constructor, or by making it an inner class, etc.)